src/commands/mod.rs

```diff
@@ -2411,6 +2438,95 @@ aborted.
     Ok(())
 }
 
+fn cmd_chmod(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &ChmodSubcommand,
+) -> Result<(), CommandError> {
+    let (executable_bit, executable_bit_str, args) = match subcommand {
+        ChmodSubcommand::Normal(args) => (false, "non-executable", args),
+        ChmodSubcommand::Executable(args) => (true, "executable", args),
+    };
+
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let repo_paths: Vec<_> = args
+        .paths
+        .iter()
+        .map(|path| workspace_command.parse_file_path(path))
+        .try_collect()?;
```

Comment:
I can't decide whether that would be helpful or indicative of a user mistake.

Reply:
That's how most (all?) other commands work anyway :) If we print "updated 92839 files", the user might realize that they probably should undo

Reply:
I'm still unsure which is better here. I initially was going to only have `chmod` act one file at a time, but it was so easy to extend it...

A minor realization I had is that `jj files | xargs jj chmod n` is a possibly safer way of achieving `jj chmod n ./` from the root. I'm unsure whether that's considered user-friendly.

Reply:
> I'm still unsure which is better here. I initially was going to only have `chmod` act one file at a time, but it was so easy to extend it...

As you can probably guess, I still think we should use a matcher for consistency with other `jj` commands :)

> A minor realization I had is that `jj files | xargs jj chmod n` is a possibly safer way of achieving `jj chmod n ./` from the root. I'm unsure whether that's considered user-friendly.

Why is that safer?

Reply:
> Why is that safer?

It doesn't require allowing `jj chmod n dir` to change many files at once, unlike `chmod -x dir`.

But sure, if you prefer using a matcher strongly enough, we can do that.

Reply:
We can also try to insure there's a message printed about many files changed even when you do `jj chmod -r some_other_change`. That seems outside the scope of this PR, I guess it would go somewhere in `tx.finish`.

Reply:
> > Why is that safer?
> 
> It doesn't require allowing `jj chmod n dir` to (potentially unexpectedly) change many files at once, unlike `chmod -x dir`.

Ah, I see what you mean.

> But sure, if you prefer using a matcher strongly enough, we can do that.

We can also leave it as is until someone actually asks for it. It's probably going to take a long time :)